### DISCUSSION TITLE
issue about Hex to Assembly format

### DISCUSSION THREAD

**Discussion by mohamadft1** [horsicq/Detect-It-Easy#137] [Q&A] [answered]
Hi
thanks a lot for your good app
I have hex code and I want to convert it to assembly code with your app, my files is .txt and .hex format and is not .exe. 
when I put this files to your app the generate assembly code is not correct an in the first of assembly code exists (3) that shouldn't to be.
can you help me for this problem?

**Comment by horsicq** [comment]
Hello! Could you please show some screenshots?

**Reply by mohamadft1** [reply]
![1](https://user-images.githubusercontent.com/32516841/201506245-92a69a30-660d-48f5-af01-0c1c7d24e31a.JPG)
![2](https://user-images.githubusercontent.com/32516841/201506249-59e9a14e-3977-4b23-b231-9870a5b786ca.JPG)

in image1 I wrote simple hex code such as 7064 in text file and select it as input of your software and then run disassembeler for 68k processors and in
 image 2  before every number your software typed 3 that in wrong and the correct hex to  assembly code conversion  of 7064 is (moveq 100, D0) that sofware give:
move.w $34(A0 D3.w), -(A3)

**Comment by horsicq** [comment, marked as answer]
You wrote "7964" in the file as text.

Text: "7964" is "37303634" in HEX format.

See the table:

![image](https://user-images.githubusercontent.com/7762949/201627058-9e2f318f-edba-4641-aeed-a73f3cbb3017.png)


If you need to disasm 7964 then you have to write it in file in HEX (not TEXT!!!)
